lib/training_module_due_date_manager.rb

```diff
@@ -47,8 +47,13 @@ def module_progress
     progress_manager.module_progress
   end
 
-  def flags
-    @tmu&.flags
+  def flags(course_id)
+    flags_hash = @tmu&.flags
+    if @training_module.exercise? && !flags_hash.nil?
```

Comment:
Instead of `!flags_hash.nil?` you can use `flags_hash.present?`

Reply:
Done @ragesoss